### DISCUSSION TITLE
Power On With Voltage

### DISCUSSION THREAD

**Discussion by git-tomb** [PiSupply/PiJuice#723] [Q&A] [answered]
Hi there,

Just wondering if there is an existing way to immediately power on the system when external power is applied?

Would be nice to include it with the Wakeup Alarm config.  First wake on external power with subsequent wakes on RTC time.

**Comment by tvoverbeek** [comment]
Exists already as a Work In Progress. See the discussion in #701.
Will be committed in the near future.
As long as the PiJuice remains powered the RTC Wakeup Alarm continues to work even without external power.

**Reply by git-tomb** [reply]
Thanks,

The 0-1% wake on charge hack doesn't do what I need. It doesn't trigger if the battery charge is already higher than 1%.

I need to power on the device immediately when the status changes from "NORMAL" (or other) to "CHARGING_FROM_IN". The reason is that I don't have access to the power button in practice.

I've looked at #701, and this could do what I want, but it doesn't actually power on the system from shutdown in its current form.  I would need the user_func for power-restored to actually send a system start signal, and I am unsure how to make that happen (does the event even trigger when the system is powered off?).

It seems user_func1.py also doesn't work. It did not create a log file when I tested the events (it is executable).

**Comment by tvoverbeek** [comment, marked as answer]
Forget what I wrote about #701.
The 'power' event in #701 is executed by the pijuice service which requires that the Pi is running, which is obviously not the case when the Pi is off.
The wakeup on charge should work also when the charge level is above 1%.

I just ran my Pi on battery and shut it down with battery level at 90% (Pi power led off and PiJuice in low power mode, wakeup on charge enabled at 1%), then when connecting external power to the PiJuice USB my Pi boots immediately.

Make sure you are running the 1.5 firmware and the 1.7 software with the Restore option for the wakeup-on-charge setting enabled. This stores the wakeup-on-charge setting in non-volatile memory.

**Reply by git-tomb** [reply]
Ah! Thanks. The Restore option is the key. I didn't see any info on that option in the Readme so never tried it.

Thanks again!

**Reply by tvoverbeek** [reply]
Yes, I know. Documentation is not updated yet. Sorry about that.
